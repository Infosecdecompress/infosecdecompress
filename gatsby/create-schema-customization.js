const createSchemaCustomization = ({ actions }) => {
  const { createTypes } = actions;
  const typeDefs = `
    type MarkdownRemark implements Node {
      frontmatter: Frontmatter
    }
    type SocialImage {
      publicURL: String
    }
    type Frontmatter {
      socialImage: SocialImage
    }
  `;

  createTypes(typeDefs);
};

module.exports = createSchemaCustomization;