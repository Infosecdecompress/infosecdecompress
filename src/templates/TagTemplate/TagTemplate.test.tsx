import React from "react";
import renderer from "react-test-renderer";

import { StaticQuery, useStaticQuery } from "gatsby";

import * as mocks from "@/mocks";

import TagTemplate from "./TagTemplate";

const mockedStaticQuery = StaticQuery as jest.Mock;
const mockedUseStaticQuery = useStaticQuery as jest.Mock;

describe("TagTemplate", () => {
  beforeEach(() => {
    mockedStaticQuery.mockImplementationOnce(({ render }) =>
      render(mocks.siteMetadata),
    );
    mockedUseStaticQuery.mockReturnValue(mocks.siteMetadata);
  });

  it("renders correctly", () => {
    const props = {
      data: {
        allMarkdownRemark: mocks.allMarkdownRemark,
      },
      pageContext: mocks.pageContext,
    };

    const tree = renderer.create(<TagTemplate {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
