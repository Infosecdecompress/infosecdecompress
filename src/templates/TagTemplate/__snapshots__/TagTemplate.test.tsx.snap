// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagTemplate renders correctly 1`] = `
<div>
  <div>
    <div>
      <div>
        <a
          href="/"
        />
        <h2>
          <a
            href="/"
          >
            John Doe
          </a>
        </h2>
        <p>
          Pellentesque odio nisi, euismod in, pharetra a, ultricies in, diam. Sed arcu.
        </p>
      </div>
      <nav>
        <ul>
          <li>
            <a
              href="/"
            >
              Articles
            </a>
          </li>
          <li>
            <a
              href="/pages/about"
            >
              About Me
            </a>
          </li>
          <li>
            <a
              href="/pages/contacts"
            >
              Contact Me
            </a>
          </li>
        </ul>
      </nav>
      <div>
        <ul>
          <li>
            <a
              href="#"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                viewBox="0 0 22 28"
              >
                <title>
                  rss
                </title>
                <path
                  d="M19.199 24C19.199 13.467 10.533 4.8 0 4.8V0c13.165 0 24 10.835 24 24h-4.801zM3.291 17.415c1.814 0 3.293 1.479 3.293 3.295 0 1.813-1.485 3.29-3.301 3.29C1.47 24 0 22.526 0 20.71s1.475-3.294 3.291-3.295zM15.909 24h-4.665c0-6.169-5.075-11.245-11.244-11.245V8.09c8.727 0 15.909 7.184 15.909 15.91z"
                />
              </svg>
            </a>
          </li>
          <li>
            <a
              href="mailto:#"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                viewBox="0 -5 38 38"
              >
                <title>
                  email
                </title>
                <path
                  d="M19,15.4615385 L36.3076923,0.461538462 L1.69230769,0.461538462 L19,15.4615385 Z M14.3251765,13.8010536 L19,17.6382399 L23.6015813,13.8010536 L36.3076923,24.6923077 L1.69230769,24.6923077 L14.3251765,13.8010536 Z M0.538461538,23.5384615 L0.538461538,1.61538462 L13.2307692,12.5769231 L0.538461538,23.5384615 Z M37.4615385,23.5384615 L37.4615385,1.61538462 L24.7692308,12.5769231 L37.4615385,23.5384615 Z"
                />
              </svg>
            </a>
          </li>
          <li>
            <a
              href="https://github.com/#"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                viewBox="0 0 26 28"
              >
                <title>
                  github
                </title>
                <path
                  d="M10 19c0 1.141-0.594 3-2 3s-2-1.859-2-3 0.594-3 2-3 2 1.859 2 3zM20 19c0 1.141-0.594 3-2 3s-2-1.859-2-3 0.594-3 2-3 2 1.859 2 3zM22.5 19c0-2.391-1.453-4.5-4-4.5-1.031 0-2.016 0.187-3.047 0.328-0.812 0.125-1.625 0.172-2.453 0.172s-1.641-0.047-2.453-0.172c-1.016-0.141-2.016-0.328-3.047-0.328-2.547 0-4 2.109-4 4.5 0 4.781 4.375 5.516 8.188 5.516h2.625c3.813 0 8.188-0.734 8.188-5.516zM26 16.25c0 1.734-0.172 3.578-0.953 5.172-2.063 4.172-7.734 4.578-11.797 4.578-4.125 0-10.141-0.359-12.281-4.578-0.797-1.578-0.969-3.437-0.969-5.172 0-2.281 0.625-4.438 2.125-6.188-0.281-0.859-0.422-1.766-0.422-2.656 0-1.172 0.266-2.344 0.797-3.406 2.469 0 4.047 1.078 5.922 2.547 1.578-0.375 3.203-0.547 4.828-0.547 1.469 0 2.953 0.156 4.375 0.5 1.859-1.453 3.437-2.5 5.875-2.5 0.531 1.062 0.797 2.234 0.797 3.406 0 0.891-0.141 1.781-0.422 2.625 1.5 1.766 2.125 3.938 2.125 6.219z"
                />
              </svg>
            </a>
          </li>
          <li>
            <a
              href="https://www.twitter.com/#"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                viewBox="-1 -1 25 25"
              >
                <title>
                  twitter
                </title>
                <path
                  d="M23.953 4.57a10 10 0 01-2.825.775 4.958 4.958 0 002.163-2.723c-.951.555-2.005.959-3.127 1.184a4.92 4.92 0 00-8.384 4.482C7.69 8.095 4.067 6.13 1.64 3.162a4.822 4.822 0 00-.666 2.475c0 1.71.87 3.213 2.188 4.096a4.904 4.904 0 01-2.228-.616v.06a4.923 4.923 0 003.946 4.827 4.996 4.996 0 01-2.212.085 4.936 4.936 0 004.604 3.417 9.867 9.867 0 01-6.102 2.105c-.39 0-.779-.023-1.17-.067a13.995 13.995 0 007.557 2.209c9.053 0 13.998-7.496 13.998-13.985 0-.21 0-.42-.015-.63A9.935 9.935 0 0024 4.59z"
                />
              </svg>
            </a>
          </li>
          <li>
            <a
              href="https://t.me/#"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                viewBox="0 0 28 28"
              >
                <title>
                  telegram
                </title>
                <path
                  d="M27.563 0.172c0.328 0.234 0.484 0.609 0.422 1l-4 24c-0.047 0.297-0.234 0.547-0.5 0.703-0.141 0.078-0.313 0.125-0.484 0.125-0.125 0-0.25-0.031-0.375-0.078l-7.078-2.891-3.781 4.609c-0.187 0.234-0.469 0.359-0.766 0.359-0.109 0-0.234-0.016-0.344-0.063-0.391-0.141-0.656-0.516-0.656-0.938v-5.453l13.5-16.547-16.703 14.453-6.172-2.531c-0.359-0.141-0.594-0.469-0.625-0.859-0.016-0.375 0.172-0.734 0.5-0.922l26-15c0.156-0.094 0.328-0.141 0.5-0.141 0.203 0 0.406 0.063 0.562 0.172z"
                />
              </svg>
            </a>
          </li>
        </ul>
      </div>
      <div>
        Content is licensed under CC BY-SA 4.0
      </div>
    </div>
  </div>
  <div>
    <div>
      <h1>
        typography
      </h1>
      <div>
        <div>
          <div>
            <div>
              <time
                dateTime="August 31, 2016"
              >
                August 2016
              </time>
              <span />
              <span>
                <a
                  href="/typography"
                >
                  typography
                </a>
              </span>
            </div>
            <h2>
              <a
                href="/posts/perfecting-the-art-of-perfection"
              >
                Perfecting the Art of Perfection
              </a>
            </h2>
            <p>
              An Essay on Typography by Eric Gill takes the reader back to the year 1930. The year when a conflict between two worlds came to its term. The machines of the industrial world finally took over the handicrafts.
            </p>
            <a
              href="/posts/perfecting-the-art-of-perfection"
            >
              Read
            </a>
          </div>
          <div>
            <div>
              <time
                dateTime="August 31, 2016"
              >
                August 2016
              </time>
              <span />
              <span>
                <a
                  href="/design-inspiration"
                >
                  design inspiration
                </a>
              </span>
            </div>
            <h2>
              <a
                href="/posts/the-birth-of-movable-type"
              >
                Johannes Gutenberg: The Birth of Movable Type
              </a>
            </h2>
            <p>
              German inventor Johannes Gutenberg developed a method of movable type and used it to create one of the western world’s first major printed books, the “Forty–Two–Line” Bible.
            </p>
            <a
              href="/posts/the-birth-of-movable-type"
            >
              Read
            </a>
          </div>
        </div>
        <div>
          <div>
            <a
              className=""
              href="/typography/page/1"
              rel="prev"
            >
              ← PREV
            </a>
          </div>
          <div>
            <a
              className=""
              href="/typography/page/3"
              rel="next"
            >
              → NEXT
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
